### DISCUSSION TITLE
Search by clicking a search button.

### DISCUSSION THREAD

**Discussion by arpitbatra123** [twilio-labs/paste#3504] [General] [answered]
In Flex v1, we supported a search input where we could trigger search by clicking on the search icon.

<img width="779" alt="image-2023-09-19-22-23-57-231" src="https://github.com/twilio-labs/paste/assets/21967563/7cb86c9d-ea71-45b6-adba-ecb0cda29ce2">

However, in Flex v2, we use an input where the Search icon is just being used as a label, and does not support "clicking' on it.

<img width="432" alt="image-2023-09-19-22-24-49-851" src="https://github.com/twilio-labs/paste/assets/21967563/65143c02-5af9-4292-ac3a-136f6e13aaa2">

I checked paste and the search input does support it If I include an icon button using  the `insertAfter` prop. 

The current input component in Flex is a custom input based on material UI, which also supports this paradigm.

Does Paste recommend using this pattern?

**Comment by TheSisb** [comment, marked as answer]
Hi @arpitbatra123,

This pattern is acceptable _for search inputs only_. You can see it in use on the [filter group pattern](https://paste.twilio.design/patterns/filter-group). The icon can be on the left in your example, that is fine too.
